cmd/clusterawsadm/cmd/ami/copy/encryptedcopy.go

```diff
@@ -0,0 +1,180 @@
+		Short: "Encrypt and copy AMI snapshot, then create an AMI with that snapshot",
+		Long: cmd.LongDesc(`
+			Find the AMI based on Kubernetes version, OS, region in the AWS account where AMIs are stored.
+			Encrypt and copy the snapshot of the AMI to the current AWS account.
+			Create an AMI with that snapshot.
+		`),
+		Example: cmd.Examples(`
+		# Create an encrypted AMI:
+		# Available os options: centos-7, ubuntu-18.04, ubuntu-20.04, amazon-2
+		clusterawsadm ami encrypted-copy --kubernetes-version=v1.18.12 --os=ubuntu-20.04  --region=us-west-2
+
+		# owner-id and dry-run flags are optional. region can be set via flag or env
+		clusterawsadm ami encrypted-copy --os centos-7 --kubernetes-version=v1.19.4 --owner-id=111111111111 --dry-run
+
+		# Encrypt using a non-default KmsKeyId specified using Key ID:
+		clusterawsadm ami encrypted-copy --os centos-7 --kubernetes-version=v1.19.4 --kms-key-id=key/1234abcd-12ab-34cd-56ef-1234567890ab
+
+		# Encrypt using a non-default KmsKeyId specified using Key alias:
+		clusterawsadm ami encrypted-copy --os centos-7 --kubernetes-version=v1.19.4 --kms-key-id=alias/ExampleAlias
+
+		# Encrypt using a non-default KmsKeyId specified using Key ARN:
+		clusterawsadm ami encrypted-copy --os centos-7 --kubernetes-version=v1.19.4 --kms-key-id=arn:aws:kms:us-east-1:012345678910:key/abcd1234-a123-456a-a12b-a123b4cd56ef
+
+		# Encrypt using a non-default KmsKeyId specified using Alias ARN:
+		clusterawsadm ami encrypted-copy --os centos-7 --kubernetes-version=v1.19.4 --kms-key-id=arn:aws:kms:us-east-1:012345678910:alias/ExampleAlias
+		`),
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			region, err := flags.GetRegion(cmd)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, "Could not resolve AWS region, define it with --region flag or as an environment variable.")
+				return err
+			}
+
+			sess, err := session.NewSessionWithOptions(session.Options{
+				SharedConfigState: session.SharedConfigEnable,
+				Config:            aws.Config{Region: aws.String(region)},
+			})
+			if err != nil {
+				fmt.Printf("Error: %v\n", err)
+				return err
+			}
+			ec2Client := ec2.New(sess)
+			dryRun, err := cmd.Flags().GetBool("dry-run")
+			if err != nil {
+				fmt.Printf("Failed to parse dry-run value: %v. Defaulting to --dry-run=false\n", err)
+			}
+
+			image, err := ec2service.DefaultAMILookup(ec2Client, ownerID, opSystem, kubernetesVersion, "")
+			if err != nil {
+				return err
+			}
+
+			if kmsKeyID == "" {
+				kmsKeyIDPtr = nil
+			}
+			kmsKeyIDPtr = &kmsKeyID
+
+			if len(image.BlockDeviceMappings) == 0 || image.BlockDeviceMappings[0].Ebs == nil {
+				return errors.New("image does not have EBS attached")
+			}
+
+			copyInput := &ec2.CopySnapshotInput{
+				Description:       image.Description,
+				DestinationRegion: &region,
+				DryRun:            &dryRun,
+				Encrypted:         pointer.BoolPtr(true),
+				SourceRegion:      &region,
+				KmsKeyId:          kmsKeyIDPtr,
```

Comment:
```suggestion
				KmsKeyId:          aws.String(kmsKeyID),
```

Reply:
If we use this, when the kmsKeyID does not exist, we don't pass nil pointer.